### PR TITLE
Use thiserror instead of anyhow

### PR DESCRIPTION
**Pull request by Furisto**
Solves #26

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ runtime = []
 
 [dependencies]
 serde = { version = "1.0.127", features = ["derive"] }
-anyhow = "1.0.42"
+thiserror = "1.0"
```
Comment (saschagrunert) [inline Cargo.toml:28]: Let's use the patch release here, too. Usually dependabot should take care of the updates.

File: src/lib.rs [hunk 2]
```diff
@@ -1,7 +1,33 @@
...
+fn oci_error<'a, M>(message: M) -> OciSpecError
+where
+    M: Into<Cow<'a, str>>,
+{
+    let message = message.into();
+    match message {
+        Cow::Borrowed(s) => OciSpecError::General(s.to_owned()),
+        Cow::Owned(s) => OciSpecError::General(s),
+    }
+}
```
Comment (saschagrunert) [inline src/lib.rs:33]: Do we wanna put this into a new `error` module?

Comment (Furisto) [reply src/lib.rs]: Sure, why not.

**Comment by Furisto** [comment]
@saschagrunert PTAL

File: src/error.rs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+use std::{borrow::Cow, io};
+use thiserror::Error;
+
+pub type Result<T> = std::result::Result<T, OciSpecError>;
+
+#[derive(Error, Debug)]
+pub enum OciSpecError {
+    #[error("{0}")]
+    General(String),
```
Comment (saschagrunert) [inline src/error.rs:9]:
```suggestion
    Other(String),
```

File: src/error.rs [hunk 4]
```diff
@@ -0,0 +1,25 @@
+use std::{borrow::Cow, io};
+use thiserror::Error;
+
+pub type Result<T> = std::result::Result<T, OciSpecError>;
+
+#[derive(Error, Debug)]
+pub enum OciSpecError {
```
Comment (saschagrunert) [inline src/error.rs:7]: Let's add docs to the types and variants since they're all public.

**Comment by Furisto** [comment]
@saschagrunert Done

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**